sotoki/sotoki.py

```diff
@@ -1138,6 +1140,22 @@ def prepare(dump_path, bin_dir):
         sys.exit("Unable to prepare xml :(")
 
 
+def check_and_optimize(path, ftype):
+    if not path.endswith(f".{ftype}"):
```

Comment:
this won't work properly for `.jpg` files. All of those will trigger this mechanism while it's expected to have JPEG files named as `.jpg`.

Reply:
Yup. I forgot handling that. Pushed a fix.